### PR TITLE
Add a translator for api/v0.2/traces protocol

### LINKED ISSUE #27045: Accept data from datadog agent
### Component(s)

receiver/datadog

### Describe the issue you're reporting

Hey, trying to collect traces from `DataDog agent` using the `OTEL collector` with the `DataDog Receiver`.
On `DataDog` side I set up another end point by adding this code to the `values.yaml` file:
```
agents:
  useConfigMap: true
  customAgentConfig:
    apm_config:
      additional_endpoints:
        "http://otel-collector-cli-service:8126":
          - "1234" 
```
On the `OTEL collector side` I added the `DataDog receiver`:
```
    receivers:
      otlp:
        protocols:
          grpc:
          http:
      datadog:
        endpoint: 0.0.0.0:8126
        read_timeout: 60s
    service:
      telemetry:
        logs:
          level: "debug"
      pipelines:
        traces:
          receivers: [otlp, datadog]
```
When a trace is created I get the following `warn` on the `DataDog trace agent`:
```
trace-agent 2023-09-19 06:53:35 UTC | TRACE | WARN | (run.go:276 in Warnf) | Stats writer payload rejected by edge: 404 Not Found
```
The `OTEL collector` log show nothing.

If the `OTEL collector` is **unreachable**, The `DataDog agent` log this message:
```
trace-agent 2023-09-21 06:55:20 UTC | TRACE | WARN | (run.go:276 in Warnf) | Retried payload 4 times: Post "http://otel-collector-cli-service:8126/api/v0.2/traces": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

Any idea what can be the problem? or if there is another way to collect `traces` from the `DataDog agent`?

### PR DESCRIPTION
**Pull request by tomert-alma**
**Description:** fixing #27045.
Fixing datadog receiver to listen to api/v0.2/traces api. 
**Link to tracking Issue:** #27045

**Testing:** add a a UT to datadog translator for  api/v0.2/traces 

This translator relies on the receiver of datadog agent:
https://github.com/DataDog/datadog-agent/blob/2973404818f6cf02138a5ae3ac9019c342adce38/test/integration/serverless/recorder-extension/main.go#L257

E2E tests has been done, where datadog receiver gets traces from the datadog agent.

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
@tomert-alma, thank you!

**Comment by tomert-alma** [comment]
@jpkrohling done. 
@MovieStoreGuy a kindly ping about this PR, can you please take a look?

File: .chloggen/datadogreceiver-api-v02-traces.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datadogereceiver
```
Comment (jpkrohling) [inline .chloggen/datadogreceiver-api-v02-traces.yaml:7]:
```suggestion
component: datadogreceiver
```

Comment (tomert-alma) [reply .chloggen/datadogreceiver-api-v02-traces.yaml]: done.

**Review by jpkrohling — APPROVED** [review]

**Comment by tomert-alma** [comment]
@jpkrohling it says that Im not authorized to merge this pull request. 
can you please merge it to main?

thanks!

**Comment by tomert-alma** [comment]
@jpkrohling a kindly ping :)  
can you please merge it to main?

**Pull request merged**